### ISSUE TITLE
Crash on startup due to different Tor Browser version schema

### ISSUE THREAD

**Issue by utzer** [Tobi823/ffupdater#514]
**Describe the bug**
FFUpdater crashes after opeing it, it shortly seaeches for updates. 

**To Reproduce**
1. Open it. 
2. It crashes. 😜

**Expected behavior**
It uptates the found apps. 

**Error message from FFUpdater**
Please copy and paste the error message with the stacktrace and error log here:

Stacktrace:
```
java.lang.IllegalStateException: Can't find download url with regex pattern '\Qhttps://dist.torproject.org/torbrowser/\E([0-9.]{4,})\Q/tor-browser-\E[0-9.]{4,}+\Q-android-aarch64-multi.apk\E'.
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser.findVersionAndDownloadUrl(TorBrowser.kt:72)
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser.access$findVersionAndDownloadUrl(TorBrowser.kt:26)
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser$findVersionAndDownloadUrl$1.invokeSuspend(TorBrowser.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7932)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:942)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@cb4178f, Dispatchers.Main]

```
Logs:
```
java.lang.IllegalStateException: Can't find download url with regex pattern '\Qhttps://dist.torproject.org/torbrowser/\E([0-9.]{4,})\Q/tor-browser-\E[0-9.]{4,}+\Q-android-aarch64-multi.apk\E'.
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser.findVersionAndDownloadUrl(TorBrowser.kt:72)
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser.access$findVersionAndDownloadUrl(TorBrowser.kt:26)
	at de.marmaro.krt.ffupdater.app.impl.TorBrowser$findVersionAndDownloadUrl$1.invokeSuspend(TorBrowser.kt:0)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7932)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:942)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@cb4178f, Dispatchers.Main]

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 78.2.4 (161) release Github |
| Device | FP4 (FP4eea, FP4, FP4) |
| Manufacturer | Fairphone (Fairphone) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 13 (SDK: 33) |
| OS | c7a8912703df, android-new, release-keys, 1693908001000 |

**Comment by utzer** [comment]
Just to be sure, it worked 2 days ago.

**Comment by jim3692** [comment]
I am also facing this issue. I think similar crashes have happened before. Would it be a good idea to show this kind of errors on the list (ex. on Tor entry show that it failed), instead of crashing the entire app?

**Comment by Usland123** [comment]
update FFUpdater to 78.2.5 version

**Comment by utzer** [comment]
it crashes during update. :-P

**Comment by Tobi823** [comment]
The bug is fixed in version 78.2.5

@utzer If you are using the FFUpdater signed by me (aka the APK from Github / Gitlab / my F-Droid repository or the developer signed APK from F-Droid) you can either:
- download the APK file from Github/Gitlab () and install it
   - https://github.com/Tobi823/ffupdater/releases/tag/78.2.5
   - https://gitlab.com/Tobiwan/ffupdater_gitlab/-/releases/78.2.5
- use F-Droid with my own repository to update FFUpdater
- or temporary uninstall Tor Browser + update FFUpdater in FFUpdater

**Comment by Tobi823** [comment]
If you are using the FFUpdater signed by F-Droid, you can either:
- switch to FFUpdater signed by me (by uninstalling FFUpdater and installing it from/with  Github / Gitlab / my F-Droid repository or the developer signed APK from F-Droid)
- wait until F-Droid published the version  78.2.5
- or temporary uninstall Tor Browser

**Comment by utzer** [comment]
@Tobi823 thanks, I added the F-Droid Repo, which fixed the update and crash problem. Thanks.
